addons/account/tests/test_account_move_entry.py

```diff
@@ -805,3 +805,34 @@ def edit_tax_on_posted_moves():
         # You can remove journal items if the related journal entry is draft.
         self.test_move.button_draft()
         edit_tax_on_posted_moves()
+
+    def test_should_allow_creating_unbalanced_misc_entry_with_tax(self):
+        # Saving an unbalanced entry isn't something desired but we need this piece of code to work in order to support
+        # the tax auto-calculation on miscellaneous move. Indeed, the JS class `AutosaveMany2ManyTagsField` triggers the
+        # saving of the record as soon as a tax base_line is modified.
+        move = self.env["account.move"].create({
+            "move_type": "entry",
+            "line_ids": [
+                Command.create({
+                    "name": "revenue line",
+                    "account_id": self.company_data["default_account_revenue"].id,
+                    'tax_ids': [Command.set(self.company_data['default_tax_sale'].ids)],
+                    "balance": -10.0,
+                }),
+            ]
+        })
+        tax_line = move.line_ids.filtered("tax_ids")
+        tax_line.unlink()
+
+        # But creating unbalanced misc entry shouldn't be allowed otherwise
+        with self.assertRaises(UserError):
```

Comment:
This could fail for other reasons and it wouldn't be detected if something unrelated changed and we are raising something else. Check a meaningful part of the error with.
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

Reply:
You're right!